sgd.lua

```diff
@@ -28,7 +28,7 @@ function optim.sgd(opfunc, x, config, state)
    -- (0) get/update state
    local config = config or {}
    local state = state or config
-   local lr = config.learningRate or 1e-3
+   local lr = config.learningRate ~= nil and config.learningRate or 1e-3
```

Comment:
Hm, I swear it was being set to 0.001 when I passed in 0. Maybe I was passing nil instead.